packages/peregrine/lib/talons/Checkout/useAddressForm.js

```diff
@@ -0,0 +1,37 @@
+import { useMemo } from 'react';
+
+/**
+ * Returns values used to render an AddressForm component.
+ * @param {Object} props
+ * @param {Object} props.fields an array of fields to reduce over for initial values
+ * @param {Object} props... TODO complete these
+ */
+export const useAddressForm = props => {
+    const {
+        onCancel,
+        countries,
+        fields,
+        initialValues,
+        isSubmitting,
+        error,
+        onSubmit
+    } = props;
+
+    const values = useMemo(
+        () =>
+            fields.reduce((acc, key) => {
+                acc[key] = initialValues[key];
+                return acc;
+            }, {}),
+        [fields, initialValues]
+    );
+
+    return {
+        countries,
+        error,
+        handleCancel: onCancel,
+        handleSubmit: onSubmit,
+        initialValues: values,
+        isSubmitting
+    };
```

Comment:
Right, good question. My gut says I should rewrite this as `isDisabled` like I did in the other form talons. This is one I missed.

Basically my process has been to pass everything back but in a "standard" way so that it's a little uniform among the talons. They all return an `isDisabled`, a `handleSubmit`, and a `handleCancel`. They probably should all take an `onSubmitError`, too. I didn't want to get too caught up in *new* API though, I would rather make this a copy-paste of most functionality and we can rework the API afterwards. Ideally we wouldn't release the unfinished API but deadlines...

Reply:
I cleaned these up. Resolving this convo but we can talk more about whether the new approach is correct when you are online.